src/features/Movies/component.js

```diff
@@ -0,0 +1,186 @@
+            ],
+			filter : 'RELEASE DATE'
+        };
+		this.handleChangeCategory = this.handleChangeCategory.bind(this);
+		this.handleChangeFilter = this.handleChangeFilter.bind(this);
+		this.handleCancelAddMovie = this.handleCancelAddMovie.bind(this);
+		this.handleChangeMovieDetails = this.handleChangeMovieDetails.bind(this);
+		this.handleSubmitMovie = this.handleSubmitMovie.bind(this);
+		this.handleResetMovie = this.handleResetMovie.bind(this);
+		this.handleEditMovie = this.handleEditMovie.bind(this);
+		this.handleRemoveMovie = this.handleRemoveMovie.bind(this);
+    }
+
+	handleCancelAddMovie(){
+		this.props.handleToggleIsHidden();
+		this.handleResetMovie();
+	}
+
+	handleSubmitMovie(e){ // Add AddMovieItem form details in the moviesToShow[] 
+		e.preventDefault();
+		this.setState({ 
+			moviesToShow : [...this.state.moviesToShow, {...this.state.movieItem, id : uuidv4()}]
+		});
+		this.props.handleToggleIsHidden();
+	}
+	
+	handleResetMovie(){ // Reset AddMovieItem form
+		this.setState({ 
+			movieItem : {name : '', date : '', category : '', rating : '', img : '', overview : '', id : ''}
+		});
+	}
+	
+	handleChangeMovieDetails(e){ // Update each form field as the user types
+		let formValue = this.state.movieItem;
+		formValue[e.target.name] = e.target.value;
+		this.setState({ movieItem : formValue });
+	}
+
+	handleEditMovie(currentMovieId){
+		this.props.handleToggleIsHidden();
+		const movieItemUpdated = this.state.moviesToShow.filter((item) => item.id === currentMovieId);
+		this.setState({ movieItem : movieItemUpdated[0] });
+	}
+
+	handleRemoveMovie(currentMovieId){
+		const moviesItemUpdated = this.state.moviesToShow.filter((item) => item.id !== currentMovieId);
+		this.setState({ moviesToShow : moviesItemUpdated });
+	}
+
+
+	handleChangeCategory(event) {
+		// change current category to active state
+		const categoriesUpdated = this.state.categories.map((category) => {
+			if (category.name === event.target.textContent){
+				return { ...category, active: true};
+			} else {
+				return {...category, active: false};
+			}
+		});
+
+		this.setState({ categories: categoriesUpdated });
+		
+		// update movies showing up according to category selected
+		this.setState((prevState) => {			
+			if (event.target.textContent === 'ALL') {
+				return { moviesToShow: prevState.movies};
+			} else {
+				const moviesUpdated = prevState.movies.filter((item) => (
+				item.category.toUpperCase().includes(event.target.textContent)
+				));	
+				return { moviesToShow: moviesUpdated};
+			}
+
+		});
+	}
+
+	handleChangeFilter(e) {
+		this.setState({ filter: `${e.target.value}` });
+		// sort movies by eityher stars or release date
+		let moviesToShowUpdated;
+
+		if (e.target.value === 'STARS'){
+			moviesToShowUpdated = this.state.moviesToShow.sort((a, b) => {
+				return b.rating - a.rating;
+			});
+		} else if (e.target.value === 'RELEASE DATE') {
+			moviesToShowUpdated = this.state.moviesToShow.sort((a, b) => {
```

Comment:
this can be an util function

Reply:
This has been now added to a util.js component